File: Arrays/Easy/1464.MaximumProduct/1464.py
from typing import List

class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        highest = -1
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if(highest < (nums[i] - 1) * (nums[j] - 1)):
                    highest = (nums[i] - 1) * (nums[j] - 1)
        return highest
    
class Solution1:
    def maxProduct(self, nums: List[int]) -> int:
        highest = nums[0]
        second = nums[1]
        if highest < second:
            highest, second = second, highest

        for i in range(2, len(nums)):
            if nums[i] > highest:
                second = highest
                highest = nums[i]
            elif nums[i] > second:
                second = nums[i]
            
        return (highest - 1) * (second - 1)
    
class Solution2:
    def maxProduct(self, nums: List[int]) -> int:
        i = max(nums)
        nums.remove(i)
        j = max(nums)
        return (i - 1) * (j - 1)

File: Arrays/Easy/1464.MaximumProduct/1464.js
var maxProduct = function(nums) {
    let highest = nums[0];
    let second = nums[1];
    if(highest < second){
        let temp = highest;
        highest = second;
        second = temp;
    }
    for(let i = 2; i < nums.length; i++){
        if(nums[i] > highest){
            second = highest;
            highest = nums[i];
        }else if(nums[i] > second){
            second = nums[i];
        }
    }
    return (highest - 1) * (second - 1);
};

var maxProduct1 = function(nums) {
    nums = nums.sort((x,y) => y-x);
    return (nums[0] - 1) * (nums[1] - 1)
};